src/librustc/hir/def.rs

```diff
@@ -93,18 +105,35 @@ pub type ExportMap = NodeMap<Vec<Export>>;
 
 #[derive(Copy, Clone, RustcEncodable, RustcDecodable)]
 pub struct Export {
-    pub name: ast::Name,    // The name of the target.
-    pub def_id: DefId, // The definition of the target.
+    pub name: ast::Name, // The name of the target.
+    pub def: Def, // The definition of the target.
```

Comment:
`DefId` alone is not enough for describing a reexport precisely.
Variants can be exported separately in type and value namespaces, but have a single `DefId` (and `NodeId`).